images/php-laravel/Dockerfile-debian

```diff
@@ -1,62 +1,53 @@
-# Use the official PHP 8.1 image with Apache as the base image
+# pull from official PHP image with Apache
 FROM php:8.1.24-apache
 
-# Install system dependencies
+# Install dependencies
 RUN apt-get update && apt-get install -y \
-    libonig-dev \
-    libpng-dev \
-    libjpeg-dev \
-    libfreetype6-dev \
-    libzip-dev \
-    zip \
-    unzip \
-    git \
-    curl
-
-# Clear out the local repository of retrieved package files
-RUN apt-get clean && rm -rf /var/lib/apt/lists/*
-
-# Install PHP extensions
-RUN docker-php-ext-install pdo_mysql mbstring zip exif pcntl \
-    && docker-php-ext-configure gd --with-freetype --with-jpeg \
-    && docker-php-ext-install gd
-
-# Get latest Composer
-COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+     libzip-dev \
+     zip \
+     supervisor \
+     && docker-php-ext-install pdo pdo_mysql zip
 
-# Set working directory
-WORKDIR /var/www/html
+# Enable Apache mod_rewrite
+RUN a2enmod rewrite
 
-# Copy existing application directory contents
+# Copy the application code to the container
 COPY . /var/www/html
 
-# Install dependencies
-RUN composer install --no-dev --optimize-autoloader
-
-# Create non root user and group for Laravel
-RUN groupadd -g 1000 laraveluser
-RUN useradd -u 1000 -ms /bin/bash -g laraveluser laraveluser
+# Set the working directory
+WORKDIR /var/www/html
 
-# Change ownership of our applications
-RUN chown -R laraveluser:laraveluser /var/www/html
+COPY /logs/laravel_err.log /var/log/laravel_err.log
```

Comment:
We'll need to configure them php.ini and apache to log to stdout.

Reply:
@gtarique  Did we get logs all going to stdout properly for apache/nginx, php, and laravel?

Reply:
@jacobbeasley yes, logs are going to stdout for apache laravel and nginx php. Sample log:

**Php/nginx:**
docker run -p 8081:8080 phpngd
2024-01-01 12:21:13,267 INFO supervisord started with pid 1
2024-01-01 12:21:14,273 INFO spawned: 'php8.0-fpm' with pid 9
2024-01-01 12:21:14,278 INFO spawned: 'nginx' with pid 10
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:1
[01-Jan-2024 12:21:14] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[01-Jan-2024 12:21:14] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[01-Jan-2024 12:21:14] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[01-Jan-2024 12:21:14] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[01-Jan-2024 12:21:14] NOTICE: fpm is running, pid 9
[01-Jan-2024 12:21:14] NOTICE: ready to handle connections
2024-01-01 12:21:15,337 INFO success: php8.0-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2024-01-01 12:21:15,337 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
NOTICE: PHP message: Test Error
127.0.0.1 -  01/Jan/2024:12:21:20 +0000 "GET /index.php" 200
192.168.65.1 - - [01/Jan/2024:12:21:20 +0000] "GET / HTTP/1.1" 200 44 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"

**PHP/Laravel:**

2024-01-01 11:52:54,775 INFO supervisord started with pid 1
2024-01-01 11:52:55,780 INFO spawned: 'apache2' with pid 7
2024-01-01 11:52:55,784 INFO spawned: 'php-fpm' with pid 8
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
[01-Jan-2024 11:52:55] NOTICE: fpm is running, pid 8
[01-Jan-2024 11:52:55] NOTICE: ready to handle connections
2024-01-01 11:52:56,837 INFO success: apache2 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2024-01-01 11:52:56,838 INFO success: php-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
192.168.65.1 - - [01/Jan/2024:11:53:00 +0000] "GET /hello HTTP/1.1" 200 1316 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
127.0.0.1 -  01/Jan/2024:11:53:00 +0000 "GET /index.php" 200
192.168.65.1 - - [01/Jan/2024:11:53:00 +0000] "GET /favicon.ico HTTP/1.1" 304 246 "http://localhost:81/hello" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"